src/elementRoleMap.js

```diff
@@ -23,7 +23,25 @@ for (let i = 0; i < keys.length; i++) {
       if (relation.module === 'HTML') {
         const concept = relation.concept;
         if (concept) {
-          elementRoles.push([concept, [key]]);
+          const elementRoleRelation: ?ElementARIARoleRelationTuple = elementRoles.find(relation => ariaRoleRelationConceptAttributeEquals(relation[0], concept));
```

Comment:
```suggestion
          const elementRoleRelation: ?ElementARIARoleRelationTuple = elementRoles.find(relation => ariaRoleRelationConceptEquals(relation[0], concept));
```

Care needed here - I _think_ you’ve been working off the wrong type(s). The `relation[0]` and `concept` are `ARIARoleRelationConcept` not `ARIARoleRelationConceptAttribute` which I think explains why your test case failure count is so high.

The comparison function needs to be amended to take the full concept into account.

Once amended the only failures should relate to the `aside` element (pending a fix for a separate issue)